### PR TITLE
Add `preserveConsecutiveUppercase` option

### LINKED ISSUE #76: `preserveConsecutiveUppercase` option
See the previous attempt and feedback in https://github.com/sindresorhus/camelcase/pull/64.

### PR DESCRIPTION
**Pull request by RocktimSaikia**
fixes #76 
Builds on the reviews left on the previous attempt #64 

- [x]  Added test cases
- [x] Updated `README`
- [x]  Synced the updated examples of readme with declaration file
- [x] Made some additional _irrelevant_ to the issue tweaks [like removing spread operators from a single property ]

This to 
```js
options = {
       ...{pascalCase: false},
       ...options
};
```
to
```js
options = {
       pascalCase: false,
       ...options
};
```

// @sindresorhus

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -71,6 +80,13 @@ Default: `false`
 
 Uppercase the first character: `foo-bar` → `FooBar`
 
+##### preserveConsecutiveUppercase
+
+Type: `boolean`\
+Default: `false`
+
+Preserve the consecutive uppercase characters: `foo-BAR` → `FooBAR`
```
Comment (sindresorhus) [inline readme.md:88]:
```suggestion
Preserve the consecutive uppercase characters: `foo-BAR` → `FooBAR`.
```

File: test.js [hunk 2]
```diff
@@ -130,6 +130,68 @@ test('camelCase with pascalCase option', t => {
...
+	t.is(camelCase('mGridCOl6@md', {preserveConsecutiveUppercase: true}), 'mGridCOl6@md');
+	t.is(camelCase('A::a', {preserveConsecutiveUppercase: true}), 'a::a');
+	t.is(camelCase('Hello1WORLD', {preserveConsecutiveUppercase: true}), 'hello1WORLD');
+	t.is(camelCase('Hello11WORLD', {preserveConsecutiveUppercase: true}), 'hello11WORLD');
+	t.is(camelCase('РозовыйПушистыйFOOдинорогиf', {preserveConsecutiveUppercase: true}), 'розовыйПушистыйFOOдинорогиf');
+	t.is(camelCase('桑德在这里。', {preserveConsecutiveUppercase: true}), '桑德在这里。');
+	t.is(camelCase('桑德_在这里。', {preserveConsecutiveUppercase: true}), '桑德在这里。');
+});
+
+test('camelCase with both -pascalCase and -preserveConsecutiveUppercase option', t => {
```
Comment (sindresorhus) [inline test.js:164]: Why the dash?

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

File: test.js (hunk 2 above)
Comment (RocktimSaikia) [reply test.js]: Oh, I just thought It would focus on the options. Never mind. That was not needed I guess.

**Comment by RocktimSaikia** [comment]
@sindresorhus Sure. I just resolved the merge conflicts. This pr can be merged now .

File: index.d.ts [hunk 3]
```diff
@@ -8,23 +8,24 @@ declare namespace camelcase {
...
+		*/
+		readonly preserveConsecutiveUppercase?: boolean;
 
+		/**
+		The locale parameter indicates the locale to be used to convert to upper/lower case according to any locale-specific case mappings. If multiple locales are given in an array, the best available locale is used.
+		Default: The host environment’s current locale.
 		@example
 		```
 		import camelCase = require('camelcase');
-
```
Comment (sindresorhus) [inline index.d.ts:18]: ?

File: index.d.ts [hunk 4]
```diff
@@ -8,23 +8,24 @@ declare namespace camelcase {
...
+		Preserve the consecutive uppercase characters: `foo-BAR` → `FooBAR`.
 
-		Default: The host environment’s current locale.
+		@default false
+		*/
+		readonly preserveConsecutiveUppercase?: boolean;
 
+		/**
+		The locale parameter indicates the locale to be used to convert to upper/lower case according to any locale-specific case mappings. If multiple locales are given in an array, the best available locale is used.
+		Default: The host environment’s current locale.
```
Comment (sindresorhus) [inline index.d.ts:19]:
```suggestion

		Default: The host environment’s current locale.

```

File: readme.md [hunk 5]
```diff
@@ -35,6 +35,12 @@ camelCase('Foo-Bar', {pascalCase: true});
 camelCase('--foo.bar', {pascalCase: false});
 //=> 'fooBar'
 
+camelCase('Foo-Bar', {preserveConsecutiveUppercase: true});
```
Comment (sindresorhus) [inline readme.md:38]: The `bar` here doesn't have consecutive uppercase characters though.

Comment (RocktimSaikia) [reply readme.md]: Ah right. Fixed it.

Comment (sindresorhus) [reply readme.md]: You need to update the type docs too.

Comment (RocktimSaikia) [reply readme.md]: Right. Just updated.

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by RocktimSaikia** [comment]
Thanks to you. I am new to open source and started contributing to open source just a few months ago only through your projects :) 
// @sindresorhus

**Pull request merged**